diff -ruN b/include/libbb.h a/include/libbb.h
--- b/include/libbb.h	2015-08-07 01:55:16.000000000 +0000
+++ a/include/libbb.h	2015-08-07 02:00:30.000000000 +0000
@@ -1836,7 +1836,7 @@
  * use bb_default_login_shell and following defines.
  * If you change LIBBB_DEFAULT_LOGIN_SHELL,
  * don't forget to change increment constant. */
-#define LIBBB_DEFAULT_LOGIN_SHELL  "-/bin/sh"
+#define LIBBB_DEFAULT_LOGIN_SHELL  "-/system/bin/sh"
 extern const char bb_default_login_shell[] ALIGN1;
 /* "/bin/sh" */
 #define DEFAULT_SHELL              (bb_default_login_shell+1)
diff -ruN b/init/init.c a/init/init.c
--- b/init/init.c	2015-08-07 01:55:16.000000000 +0000
+++ a/init/init.c	2015-08-07 02:00:30.000000000 +0000
@@ -1067,7 +1067,7 @@
 	/* Make sure environs is set to something sane */
 	putenv((char *) "HOME=/");
 	putenv((char *) bb_PATH_root_path);
-	putenv((char *) "SHELL=/bin/sh");
+	putenv((char *) "SHELL=/system/bin/sh");
 	putenv((char *) "USER=root"); /* needed? why? */
 
 	if (argv[1])
diff -ruN b/shell/ash.c a/shell/ash.c
--- b/shell/ash.c	2015-08-07 01:55:16.000000000 +0000
+++ a/shell/ash.c	2015-08-07 02:01:46.000000000 +0000
@@ -13167,11 +13167,13 @@
 
 	if (argv[0] && argv[0][0] == '-')
 		isloginsh = 1;
+	else
+		isloginsh = 1;
 	if (isloginsh) {
 		const char *hp;
 
 		state = 1;
-		read_profile("/etc/profile");
+		read_profile("/system/etc/profile");
  state1:
 		state = 2;
 		hp = lookupvar("HOME");
@@ -13212,13 +13214,16 @@
 		if (iflag) {
 			const char *hp = lookupvar("HISTFILE");
 			if (!hp) {
+/*
 				hp = lookupvar("HOME");
 				if (hp) {
 					hp = concat_path_file(hp, ".ash_history");
 					setvar0("HISTFILE", hp);
 					free((char*)hp);
 					hp = lookupvar("HISTFILE");
+*/
 				}
+				setvar("HISTFILE", "/data/ash_history", 0);
 			}
 			if (hp)
 				line_input_state->hist_file = hp;
